gammapy/modeling/models/prior.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import numpy as np
+import astropy.units as u
+from gammapy.modeling import PriorParameter, PriorParameters
+from .core import ModelBase
+
+log = logging.getLogger(__name__)
+
+
+def _build_priorparameters_from_dict(data, default_parameters):
+    """Build Parameters object from input dict and default parameter values."""
+    par_data = []
+
+    input_names = [_["name"] for _ in data]
+
+    for par in default_parameters:
+        par_dict = par.to_dict()
+        try:
+            index = input_names.index(par_dict["name"])
+            par_dict.update(data[index])
+        except ValueError:
+            log.warning(
+                f"PriorParameter '{par_dict['name']}' not defined in YAML file."
+                f" Using default value: {par_dict['value']} {par_dict['unit']}"
+            )
+        par_data.append(par_dict)
+
+    return PriorParameters.from_dict(par_data)
+
+
+class PriorModel(ModelBase):
+    _weight = 1
```

Comment:
should this be a parameter ? at least it should be serialised

Reply:
Yes, you are right! The weight is now a parameter which can be set with or after the initialisation and it gets serialised.